packages/devtools_app/test/support/mocks.dart

```diff
@@ -15,8 +19,269 @@ class MockServiceManager extends Mock implements ServiceConnectionManager {
 
   @override
   final ConnectedApp connectedApp = MockConnectedApp();
+
+  @override
+  final FakeServiceExtensionManager serviceExtensionManager =
+      FakeServiceExtensionManager();
 }
 
 class MockVmService extends Mock implements VmServiceWrapper {}
 
 class MockConnectedApp extends Mock implements ConnectedApp {}
+
+/// Fake that simplifies writing UI tests that depend on the
+/// ServiceExtensionManager.
+class FakeServiceExtensionManager extends Fake
```

Comment:
I'm sure you've already done as much pruning as you can, but it seems like there is a lot of duplicated code in this class. So just a note that we should try to strip as much of this out as possible if not all of it is necessary for the fake.

Reply:
yeah. this code isn't great but was quick to write. Added a TODO to do a cleaner refactor in the future if we want to write more tests of this functionality.